portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -61,7 +68,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         break;
       }
       String content = (String) entity.getProperty(CONTENT);
-      messages.add(content);
+      try {
+        Translation translation = translate.translate(content, Translate.TranslateOption.targetLanguage(language));
```

Comment:
This is sending an rpc for each string to translate. It would be better to send just one rpc for all strings to translate. You can use:

`translate(List<String> texts, Translate.TranslateOption... options)`

Documentation here: https://googleapis.dev/java/google-cloud-translate/latest/com/google/cloud/translate/Translate.html#translate-java.util.List-com.google.cloud.translate.Translate.TranslateOption...-

Reply:
> 
> 
> Do you have a demo link?

Currently the demo is not working with a server 500 error, I will post it when it's fixed.